packages/ref-core/tests/unit/cmec/test_cmec_metric.py

```diff
@@ -0,0 +1,347 @@
+    cmec_right_metric_dict["RESULTS"]["attributes"] = "some information for results level"
+
+    with pytest.raises(ValidationError):
+        CMECMetric(**cmec_right_metric_dict)
+
+
+def test_validate_result_wo_dim(cmec_right_metric_dict):
+    with pytest.raises(ValidationError):
+        MetricResults(cmec_right_metric_dict["RESULTS"])
+
+
+def test_metric_merge():
+    import json
+
+    dict_pmp = {
+        "DIMENSIONS": {
+            "json_structure": ["model", "metric"],
+            "model": {"GFDL-ESM2M": {"Source": "CMIP5 ESGF"}},
+            "metric": {
+                "NinoSstDiversity_2": {"Name": "NinoSstDiversity_2"},
+                "BiasTauxLonRmse": {"name": "BiasTauxLonRmse"},
+            },
+        },
+        "RESULTS": {
+            "GFDL-ESM2M": {
+                "NinoSstDiversity_2": -75,
+                "BiasTauxLonRmse": 5.959564210237568,
+            }
+        },
+    }
+    dict_ilamb = {
+        "DIMENSIONS": {
+            "json_structure": ["model", "metric"],
+            "model": {"E3SM": {"name": "E3SM"}, "CESM": {"name": "CESM"}},
+            "metric": {"carbon": {"name": "carbon"}},
+        },
+        "RESULTS": {
+            "E3SM": {"carbon": {"overall score": 0.11, "bias": 0.56}},
+            "CESM": {"carbon": {"overall score": 0.05, "bias": 0.72}},
+        },
+    }
+
+    dict_merged = {
+        "SCHEMA": None,
+        "DIMENSIONS": {
+            "json_structure": ["model", "metric"],
+            "model": {
+                "GFDL-ESM2M": {"Source": "CMIP5 ESGF"},
+                "E3SM": {"name": "E3SM"},
+                "CESM": {"name": "CESM"},
+            },
+            "metric": {
+                "NinoSstDiversity_2": {"Name": "NinoSstDiversity_2"},
+                "BiasTauxLonRmse": {"name": "BiasTauxLonRmse"},
+                "carbon": {"name": "carbon"},
+            },
+        },
+        "RESULTS": {
+            "E3SM": {
+                "carbon": {"overall score": 0.11, "bias": 0.56},
+                "NinoSstDiversity_2": {},
+                "BiasTauxLonRmse": {},
+            },
+            "CESM": {
+                "carbon": {"overall score": 0.05, "bias": 0.72},
+                "NinoSstDiversity_2": {},
+                "BiasTauxLonRmse": {},
+            },
+            "GFDL-ESM2M": {
+                "NinoSstDiversity_2": -75,
+                "BiasTauxLonRmse": 5.959564210237568,
+                "carbon": {},
+            },
+        },
+        "PROVENANCE": None,
+        "DISCLAIMER": None,
+        "NOTES": None,
+    }
+
+    assert (
+        json.loads(CMECMetric.merge(dict_pmp, dict_ilamb, -9999.0).model_dump_json(indent=2)) == dict_merged
+    )
+
+
+def test_gen_json_schema():
+    from cmip_ref_core.pycmec.metric import (
+        CMECGenerateJsonSchema,
+    )
+
+    cmec_model_schema = CMECMetric.model_json_schema(schema_generator=CMECGenerateJsonSchema)
+
+    assert cmec_model_schema == {
```

Comment:
You might want to use pytest-regressions to track this schema as a separate file. I find that easier to maintain than inlining for things that are generated.

See the tests in packages/ref-core/tests/unit/test_datasets.py which use the `data_regression` and the docs here; https://pytest-regressions.readthedocs.io/en/latest/overview.html

Reply:
Cool. Thanks for the suggestion. I will follow it.